src/controllers/apps/video-app/comment.controller.js

```diff
@@ -0,0 +1,256 @@
+import { asyncHandler } from "../../../utils/asyncHandler.js";
+import { Video } from "../../../models/apps/video-app/video.model.js";
+import { getMongoosePaginationOptions } from "../../../utils/helpers.js";
+
+/**
+ * Retrieves the comments for a specific video.
+ *
+ * @param {Object} req - The request object.
+ * @param {Object} res - The response object.
+ * @returns {Object} - The response object containing the video comments.
+ * @throws {ApiError} - If the video comments are not found.
+ */
+const getVideoComments = asyncHandler(async (req, res) => {
+  try {
+    const { videoId } = req.params;
+
+    if (!isValidObjectId(videoId)) {
+      throw new ApiError(400, "Invalid Id");
+    }
+
+    const { page = 1, limit = 10 } = req.query;
+
+    const videosCommentAggregate = Video.aggregate([
+      {
+        $match: {
+          _id: new mongoose.Types.ObjectId(videoId),
+        },
+      },
+      {
+        $lookup: {
+          from: "comments",
+          localField: "_id",
+          foreignField: "video",
+          as: "comments",
+          pipeline: [
+            {
+              $sort: {
+                createdAt: -1,
+              },
+            },
+          ],
+        },
+      },
+      {
+        $unwind: {
+          path: "$comments",
+        },
+      },
+      {
+        $lookup: {
+          from: "users",
+          localField: "comments.owner",
+          foreignField: "_id",
+          as: "comments.owner",
+        },
+      },
+
+      {
+        $unwind: {
+          path: "$comments.owner",
+        },
+      },
+      {
+        $group: {
+          _id: "$_id",
+          title: { $first: "$title" },
+          description: { $first: "$description" },
+          thumbnail: { $first: "$thumbnail" },
+          videoFile: { $first: "$videoFile" },
+          owner: { $first: "$owner" },
+          views: { $first: "$views" },
+          createdAt: { $first: "$createdAt" },
+          updatedAt: { $first: "$updatedAt" },
+          comments: { $push: "$comments" },
+        },
+      },
+
+      {
+        $project: {
+          _id: 1,
+          title: 1,
+          description: 1,
+          thumbnail: 1,
+          videoFile: 1,
+          owner: 1,
+          views: 1,
+          createdAt: 1,
+          updatedAt: 1,
+          comments: {
+            _id: 1,
+            content: 1,
+            owner: {
+              _id: 1,
+              username: 1,
+              email: 1,
+              avatar: 1,
+              fullname: 1,
+            },
+            createdAt: 1,
+            updatedAt: 1,
+          },
+        },
+      },
+    ]);
+
+    const videos = await Video.aggregatePaginate(
+      videosCommentAggregate,
+      getMongoosePaginationOptions({
+        page,
+        limit,
+        customLabels: {
+          docs: "videosComment",
+          totalDocs: "totalVideosComment",
+        },
+      })
+    );
+
+    if (!videos) {
```

Comment:
This condition seems wrong. vidoes will always be an object which is a truthy value so this condition seems redundant.